From 2258db135541057abe496c29cecdf84d096d7120 Mon Sep 17 00:00:00 2001
From: "tao.dong" <tao.dong@amlogic.com>
Date: Fri, 18 Jul 2014 15:55:03 +0800
Subject: [PATCH 4539/5965] PD#93541:remove unused warning commit
 a5a7c99fb171d8b211ee60aafbeb7a61f4a0b68a Author: xu xing
 <xing.xu@amlogic.com> Date:   Mon Jul 14 16:52:45 2014 +0800

    PD #93541:Remove [-Wunused-value] type warning

commit a6d8ea840918d85c4863803a1ab9828946f2e3da
Author: xu xing <xing.xu@amlogic.com>
Date:   Mon Jul 14 15:59:36 2014 +0800

    PD #93541:Remove [-Wmissing-braces] type warning

commit 43a6dc029a3a517e9faa90255cf5b596f5a2ad38
Author: xu xing <xing.xu@amlogic.com>
Date:   Mon Jul 14 15:15:11 2014 +0800

    pd #93541:Remove [-Wimplicit-int] type warning

commit ec268575edd428f1b85d70b553b35e26bdb4d0ab
Author: xu xing <xing.xu@amlogic.com>
Date:   Mon Jul 14 14:49:10 2014 +0800

    PD #93541:Remove [-Wcomment] type warning and enable control flag.

commit 4d8a2a51f42876658dbefa7c1613da043fe49497
Author: xu xing <xing.xu@amlogic.com>
Date:   Mon Jul 14 14:21:24 2014 +0800

    PD #93541: Remove [-Wenum-compare] type warning and enable control flag
---
 Makefile                                      |  5 ++
 arch/arm/plat-meson/debug.c                   |  2 +-
 drivers/amlogic/camera/ov5647.c               |  2 +-
 drivers/amlogic/ethernet/am_mdio.c            |  4 +-
 drivers/amlogic/ethernet/phy/am_internal.c    | 52 +++++++++----------
 .../touchscreen/gslx680/amlogic_gslX680.c     |  3 +-
 drivers/amlogic/video_dev/amlvideo.c          |  4 +-
 7 files changed, 37 insertions(+), 35 deletions(-)

diff --git a/Makefile b/Makefile
index 6d34914f9324..514b879cdc3a 100755
--- a/Makefile
+++ b/Makefile
@@ -374,6 +374,11 @@ KBUILD_CFLAGS   := -Wall -Wundef -Wstrict-prototypes -Wno-trigraphs \
 		   -Werror-implicit-function-declaration \
 		   -Wno-format-security \
 		   -fno-delete-null-pointer-checks
+KBUILD_CFLAGS   += -Werror=enum-compare \
+		   -Werror=comment \
+		   -Werror=implicit-int \
+		   -Werror=missing-braces \
+		   -Werror=unused-value
 KBUILD_AFLAGS_KERNEL :=
 KBUILD_CFLAGS_KERNEL :=
 KBUILD_AFLAGS   := -D__ASSEMBLY__
diff --git a/arch/arm/plat-meson/debug.c b/arch/arm/plat-meson/debug.c
index 53261c5f88f6..24c1c36cd087 100755
--- a/arch/arm/plat-meson/debug.c
+++ b/arch/arm/plat-meson/debug.c
@@ -328,7 +328,7 @@ int do_gpio_test_work(char argn ,char **argv )
 
 #define MAX_ARG_NUM 4
 
-#endif /*CONFIG_GPIO_TEST/*
+#endif /*CONFIG_GPIO_TEST*/
 
 
 
diff --git a/drivers/amlogic/camera/ov5647.c b/drivers/amlogic/camera/ov5647.c
index 2d028e443305..402989e3cdec 100755
--- a/drivers/amlogic/camera/ov5647.c
+++ b/drivers/amlogic/camera/ov5647.c
@@ -2824,7 +2824,7 @@ void OV5647_set_param_effect(struct ov5647_device *dev,enum camera_effect_flip_e
     int index = 0;
     int i = 0;
     while(i < ARRAY_SIZE(effect_pair)){
-        if(effect_pair[i].effect == para){
+        if((unsigned int)effect_pair[i].effect == (unsigned int)para){
             break;
         }else
             i++;
diff --git a/drivers/amlogic/ethernet/am_mdio.c b/drivers/amlogic/ethernet/am_mdio.c
index ab8edb215c24..56ab2625d2e9 100755
--- a/drivers/amlogic/ethernet/am_mdio.c
+++ b/drivers/amlogic/ethernet/am_mdio.c
@@ -107,10 +107,8 @@ int aml_mdio_register(struct net_device *ndev)
 			}
 			if (phydev->phy_id  != 0) {
 				//priv->phydev->addr = addr;
-				if ((phydev->phy_id  == 0x001cc916)&& (addr == 0)) 
+				if (!((phydev->phy_id  == 0x001cc916)&& (addr == 0))) 
 				{
-					NULL;
-				}else{
 					priv->phy_addr = addr;
 					phydev->irq = PHY_POLL;
 					irqlist[addr] = PHY_POLL;
diff --git a/drivers/amlogic/ethernet/phy/am_internal.c b/drivers/amlogic/ethernet/phy/am_internal.c
index 8b5f3382d73c..697b96e59557 100755
--- a/drivers/amlogic/ethernet/phy/am_internal.c
+++ b/drivers/amlogic/ethernet/phy/am_internal.c
@@ -243,34 +243,32 @@ static int internal_read_status(struct phy_device *phydev)
 	if(phydev->speed == SPEED_100){
 		init_internal_phy_100B(phydev);
 	}
-	if (AUTONEG_ENABLE == phydev->autoneg){
-		NULL;
+	if (!(AUTONEG_ENABLE == phydev->autoneg)){
+		if (!phydev->link) {
+			/* Disable EDPD to wake up PHY */
+			int rc = phy_read(phydev, MII_INTERNAL_CTRL_STATUS);
+			if (rc < 0)
+				return rc;
+
+			rc = phy_write(phydev, MII_INTERNAL_CTRL_STATUS,
+					rc & ~MII_INTERNAL_EDPWRDOWN);
+			if (rc < 0)
+				return rc;
+
+			/* Sleep 64 ms to allow ~5 link test pulses to be sent */
+			msleep(64);
+
+			/* Re-enable EDPD */
+			rc = phy_read(phydev, MII_INTERNAL_CTRL_STATUS);
+			if (rc < 0)
+				return rc;
+
+			rc = phy_write(phydev, MII_INTERNAL_CTRL_STATUS,
+					rc | MII_INTERNAL_EDPWRDOWN);
+			if (rc < 0)
+				return rc;
+		}
 	}
-	else if (!phydev->link) {
-		/* Disable EDPD to wake up PHY */
-		int rc = phy_read(phydev, MII_INTERNAL_CTRL_STATUS);
-		if (rc < 0)
-			return rc;
-
-		rc = phy_write(phydev, MII_INTERNAL_CTRL_STATUS,
-				rc & ~MII_INTERNAL_EDPWRDOWN);
-		if (rc < 0)
-			return rc;
-
-		/* Sleep 64 ms to allow ~5 link test pulses to be sent */
-		msleep(64);
-
-		/* Re-enable EDPD */
-		rc = phy_read(phydev, MII_INTERNAL_CTRL_STATUS);
-		if (rc < 0)
-			return rc;
-
-		rc = phy_write(phydev, MII_INTERNAL_CTRL_STATUS,
-				rc | MII_INTERNAL_EDPWRDOWN);
-		if (rc < 0)
-			return rc;
-	}
-
 	return err;
 }
 
diff --git a/drivers/amlogic/input/touchscreen/gslx680/amlogic_gslX680.c b/drivers/amlogic/input/touchscreen/gslx680/amlogic_gslX680.c
index 5cb7d4100c60..b615e0ae8319 100755
--- a/drivers/amlogic/input/touchscreen/gslx680/amlogic_gslX680.c
+++ b/drivers/amlogic/input/touchscreen/gslx680/amlogic_gslX680.c
@@ -710,7 +710,8 @@ static void gslX680_ts_worker(struct work_struct *work)
 #ifdef GSL_NOID_VERSION
 	u32 tmp1;
 	u8 buf[4] = {0};
-	struct gsl_touch_info cinfo = {0};
+	struct gsl_touch_info cinfo;
+	memset(&cinfo,0,sizeof(struct gsl_touch_info));
 #endif
 
 	rc = gsl_ts_read(ts->client, 0x80, ts->touch_data, ts->dd->data_size);
diff --git a/drivers/amlogic/video_dev/amlvideo.c b/drivers/amlogic/video_dev/amlvideo.c
index 2ce9663df2b9..b91441a0b003 100755
--- a/drivers/amlogic/video_dev/amlvideo.c
+++ b/drivers/amlogic/video_dev/amlvideo.c
@@ -60,7 +60,7 @@
 
 #define AMLVIDEO_POOL_SIZE 16
 static vfq_t q_ready;
-extern omx_secret_mode;
+extern bool omx_secret_mode;
 static u8 first_frame;
 static u32 vpts_last;
 
@@ -629,7 +629,7 @@ static int freerun_dqbuf(struct v4l2_buffer *p) {
 }
 
 static int normal_dqbuf(struct v4l2_buffer *p) {
-    static last_pcrscr = 0;
+    static int last_pcrscr = 0;
     int ret = 0;
     int diff = 0;
     int a = 0;
-- 
2.19.0

